app/models/entities/item.rb

```diff
@@ -28,106 +36,143 @@ def consolidate_and_map_data(connec_entities, external_entities, organization, o
 
   def push_entities_to_connec(connec_client, mapped_external_entities_with_idmaps, organization)
     # 1/ push the orders
-    self.push_entities_to_connec_to(connec_client, mapped_external_entities_with_idmaps, self.connec_entity_name, organization)
+    self.push_entities_to_connec_to(connec_client, mapped_external_entities_with_idmaps, self.class.connec_entity_name, organization)
     variants = []
     mapped_external_entities_with_idmaps.each do |mapped_external_entities_with_idmap|
       parent_connect_id = mapped_external_entities_with_idmap[:idmap].connec_id
       product = mapped_external_entities_with_idmap[:entity]
       product[:variants].each do |variant|
         variant[:parent_item_id] = parent_connect_id
-        idmap = Maestrano::Connector::Rails::IdMap.find_by(external_id: variant[:external_id], connec_entity: connec_entity_name.downcase, external_entity: 'variant', organization_id: organization.id)
-        variants.push({entity: variant, idmap: idmap || create_id_map(variant, organization)})
+        idmap = Maestrano::Connector::Rails::IdMap.find_by(external_id: variant[:external_id], connec_entity: self.class.connec_entity_name.downcase, external_entity: 'variant', organization_id: organization.id)
+        variants.push({entity: variant, idmap: idmap || create_variant_id_map(variant, organization)})
       end if product[:variants]
```

Comment:
not so fan of the if here
would rather have put a `next unless product[:variants]` before the loop